src/posix/platform/hdlc_interface.cpp

```diff
@@ -656,6 +667,94 @@ void HdlcInterface::HandleHdlcFrame(otError aError)
     }
 }
 
+otError HdlcInterface::ResetSocket(void)
+{
+    otError error = OT_ERROR_NONE;
+
+    SuccessOrExit(error = WaitForUsbDeviceRenumeration(mRadioUrl.GetPath()));
+
+    CloseFile();
+
+    mSockFd = OpenFile(mRadioUrl);
+    VerifyOrExit(mSockFd != -1, error = OT_ERROR_FAILED);
+
+exit:
+    return error;
+}
+
+otError HdlcInterface::WaitForUsbDeviceRenumeration(const char *aRadioUrlPath)
+{
+    otError error = OT_ERROR_NONE;
+
+#ifdef OPENTHREAD_USE_UDEV
+    struct udev *udev = udev_new();
+    if (!udev)
+    {
+        otLogCritPlat("Can't create udev");
+        return OT_ERROR_FAILED;
```

Comment:
From OT style guide we aim to have a single `return` from any method/function (at the end of it). 
Suggest suing `VerifyOrExit()` or other macros for exiting early.